File: KMMClient/shared/src/commonMain/kotlin/co/touchlab/kampkit/transformer/ParentPlatformTransformer.kt
package co.touchlab.kampkit.transformer

import co.touchlab.kampkit.data.local.entity.ParentPlatformEntity
import co.touchlab.kampkit.data.remote.response.GameParentPlatformResponse
import co.touchlab.kampkit.data.remote.response.ParentPlatformResponse
import co.touchlab.kampkit.domain.model.ParentPlatformModel

fun ParentPlatformResponse.toEntity() = ParentPlatformEntity(id, name, slug)
fun List<ParentPlatformResponse>.toEntities() = map { it.toEntity() }

fun ParentPlatformEntity.toDomainModel() = ParentPlatformModel(id, name, slug)
fun List<ParentPlatformEntity>.toDomainModels() = map { it.toDomainModel() }

File: iOSNative/Games/Sources/Games/Transformer/ParentPlatformTransformer.swift
//
//  File.swift
//  
//
//  Created by Abdul Chathil on 31/03/21.
//

import Foundation

extension Array where Element == GameParentPlatformResponse {
  func toEntities() -> [ParentPlatformEntity] {
    map { $0.platform.toEntity() }
  }
}

extension ParentPlatformResponse {
  func toEntity() -> ParentPlatformEntity {
    ParentPlatformEntity(id: id, name: name, slug: slug)
  }
}
